admin/create-theme/theme-utils.php

```diff
@@ -185,4 +185,55 @@ public static function is_valid_screenshot( $file ) {
 		return 0;
 	}
 
+	public static function is_valid_screenshot_file( $file_path ) {
+		return Theme_Utils::get_screenshot_file_extension( $file_path ) !== null;
+	}
+
+	public static function get_screenshot_file_extension( $file_path ) {
+		$allowed_screenshot_types = array(
+			'png'  => 'image/png',
+			'gif'  => 'image/gif',
+			'jpg'  => 'image/jpeg',
+			'jpeg' => 'image/jpeg',
+			'webp' => 'image/webp',
+			'avif' => 'image/avif',
+		);
+		$filetype                 = wp_check_filetype( $file_path, $allowed_screenshot_types );
+		if ( in_array( $filetype['type'], $allowed_screenshot_types, true ) ) {
+			return $filetype['ext'];
+		}
+		return null;
+	}
+
+	public static function process_screenshot( $file_path ) {
+		$new_screeenshot_id      = attachment_url_to_postid( $file_path );
```

Comment:
`attachment_url_to_postid()` function [returns zero when no post ID was found](https://github.com/WordPress/wordpress-develop/blob/08b4dd3b8b96ca388b939bb049f3177b90c2d883/src/wp-includes/media.php#L5316). Just in case, how about adding some error checking here?

Reply:
Defensive logic added.